### PR TITLE
CMOS-5275 Remove moment use from plasma

### PR DESCRIPTION
**Pull request by maniator**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,7 +77,7 @@
     "eslint-plugin-prettier": "3.0.1",
     "eslint-plugin-react": "6.10.3",
     "extract-text-webpack-plugin": "3.0.2",
-    "flow-bin": "0.105.2",
+    "flow-bin": "^0.107.0",
```
Comment (kangax) [inline package.json]: we should pin all deps

File: src/components/TimePicker/TimePicker.jsx [hunk 2]
```diff
@@ -90,27 +97,30 @@ type State = {|
   options: Array<TimeOption>,
 |};
 
-const moment24h = (value?: string): moment => moment(value, FormatTypes.timeFormat24);
+const isBefore = (date, otherDate) => date.getTime() < otherDate.getTime();
```
Comment (kangax) [inline src/components/TimePicker/TimePicker.jsx]: let's type-annotate these args

File: package.json (hunk 1 above)
Comment (maniator) [reply package.json]: Done

File: src/components/TimePicker/TimePicker.jsx (hunk 2 above)
Comment (maniator) [reply src/components/TimePicker/TimePicker.jsx]: Done.

File: src/components/TimePicker/TimePicker.jsx [hunk 3]
```diff
@@ -194,15 +204,17 @@ class TimePicker extends React.Component<Props, State> {
...
-        const roundedUp = Math.ceil(moment().minute() / timeIntervalMinutes) * timeIntervalMinutes;
+        const now = new Date();
+        const roundedUp = Math.ceil(now.getMinutes() / timeIntervalMinutes) * timeIntervalMinutes;
 
-        return moment()
-          .minute(roundedUp)
-          .second(0)
-          .format(TimePicker.FormatTypes.timeFormat24);
+        const defaultTime = new Date(now);
+        defaultTime.setMinutes(roundedUp);
```
Comment (pseuyi) [inline src/components/TimePicker/TimePicker.jsx]: can be chained?

Comment (maniator) [reply src/components/TimePicker/TimePicker.jsx]: Nope. I tried :-( 

it returns the new minutes as an int

**Pull request merged**